src/libraries/Microsoft.PowerFx.Interpreter/UDF/CheckWrapper.cs

```diff
@@ -16,14 +16,20 @@ internal class CheckWrapper
         private readonly string _expressionText;
         private readonly RecordType _parameterType;
         private readonly RecalcEngine _engine;
+        public readonly ParserOptions ParserOptions;
 
-        public CheckWrapper(RecalcEngine engine, string expressionText, RecordType parameterType = null)
+        public CheckWrapper(RecalcEngine engine, string expressionText, RecordType parameterType = null, bool isImperative = false)
         {
             _engine = engine;
             _expressionText = expressionText;
             _parameterType = parameterType;
+            ParserOptions = new ParserOptions();
+
+            // If culture isn't explicitly set, use the one from PowerFx Config
+            ParserOptions.Culture ??= _engine.Config.CultureInfo;
```

Comment:
nit: can be an initialization body on ParserOptions:
`
ParserOptions = new ParserOptions() 
{
Culture = _engine.Config.CultureInfo,
AllowsSideEffects = isImperative
}